pinot-spi/src/test/java/org/apache/pinot/spi/env/PinotConfigurationTest.java

```diff
@@ -24,24 +24,116 @@
 import java.nio.file.Files;
 import java.nio.file.Paths;
 import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.apache.pinot.spi.ingestion.batch.spec.PinotFSSpec;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 
 public class PinotConfigurationTest {
+
+  @Test
+  public void assertBaseOperations() {
```

Comment:
Thanks for adding these unit tests. Wondering if we can also exercise the various code paths that perform different subsettings (eg for PinotFS, Crypter, Server, etc) with tests?

Reply:
Pleasure is mine @mayankshriv, 

Line 116 of `PinotConfiguration` executes some tests on config subsettings. Kishore added tests specifically for PinotFS, any additional contextual tests related to Crypter and Server should be hosted in their dedicated modules. That would be a great addition but maybe should be tied to a future PR that aimed at adding tests for these specific modules. What do you thing?

Reply:
I acknowledge that lack of coverage was an issue prior to the PinotConfiguration change. However, given that this change has a bigger impact, we run the risk of running into issues each time a specific setup (eg Crypter) is exercised. I was proposing to simply check the config part of those modules working.

Ideally, it would have been great to have those tests to begin with. But in the absence of those (and now that we have run into an issue), the next best thing would have been adding tests to ensure the coverage before the PinotConfiguration refactor.

Open to suggestion if you have any, on avoiding running into issues and fixing them as we find them.

Reply:
@jackjlli ^^